### DISCUSSION TITLE
Is it possible to define custom handler?

### DISCUSSION THREAD

**Discussion by emil14** [tliron/glsp#31] [General]
Fix me if I'm wrong but on a client I can do `client.sendRequest("foobar", {})` which mean it should be possible to defined custom handler. But looking at `Handler` interface implementation in the lib it looks like there's a switch that simply do return in case there is no match with predefined paths. Or Am I wrong?

**Comment by semanticart** [comment]
I'm also curious about this. Non-standard methods not super uncommon (the ruby-lsp recently added a method for getting dependencies, for instance).

**Comment by tliron** [comment]
Hm, it's honestly something I did not know. I can think of a potential design for this, but very happy to review a PR to add this!

**Comment by tris203** [comment]
I am working on an implementation where I would also like this. I would be happy to work on a PR. Do you have a breif overview of the design you were thinking of for this @tliron just to make sure we are aligned?

**Reply by tliron** [reply]
I don't have a design ready right now, but you can do the opening shot and share your idea.

**Reply by tris203** [reply]
I have started an implementation this evening. I have the basis of it working, but just having an issue with the interface for the params as its so generic as we dont know what we are getting. Unmarshalling it in the current system is tricky

**Reply by tris203** [reply]
Initial design in #32

**Comment by tliron** [comment]
How about we turn this issue into a discussion and share design ideas?

**Comment by tris203** [comment]
> How about we turn this issue into a discussion and share design ideas?

Sure

**Comment by zaibon** [comment]
Hi 👋 
I'm curious to see what's the status of this feature? I see [the PR](https://github.com/tliron/glsp/pull/32) is actually ready. Is there any blockers to move forward to merge it?

**Comment by tliron** [comment]
The only blocker is time. :) I am considering looking for a new maintainer for the project... Will try to get to it this week.
